metricbeat/module/kafka/partition/partition_integration_test.go

```diff
@@ -61,15 +60,13 @@ func TestData(t *testing.T) {
 }
 
 func TestTopic(t *testing.T) {
-	t.Skip("Flaky test: https://github.com/elastic/beats/issues/42808")
 	service := compose.EnsureUp(t, "kafka",
 		compose.UpWithTimeout(600*time.Second),
 		compose.UpWithAdvertisedHostEnvFileForPort(9092),
 	)
 
 	logp.TestingSetup(logp.WithSelectors("kafka"))
-
-	id := strconv.Itoa(rand.New(rand.NewSource(int64(time.Now().Nanosecond()))).Int())
+	id := strconv.Itoa(rand.New(rand.NewPCG(rand.Uint64(), rand.Uint64())).Int())
```

Comment:
It is not necessary to seed a new source here; the global source is already randomised at [init](https://cs.opensource.google/go/go/+/refs/tags/go1.24.2:src/math/rand/v2/rand.go;l=254-266) in the [runtime](https://cs.opensource.google/go/go/+/refs/tags/go1.24.2:src/runtime/rand.go;l=153-180).
```suggestion
	id := strconv.Itoa(rand.Int())
```